include/crypto/asn1.h

```diff
@@ -7,7 +7,11 @@
  * https://www.openssl.org/source/license.html
  */
 
-#include <openssl/asn1.h>
+#ifndef OSSL_CRYPTO_ASN1_H
```

Comment:
Are these guards helpful for internal header files?
We shouldn't be including them more than once.

Reply:
They are not necessary if it's guaranteed that they are included only once, but they don't do any harm either. But we should try to be consistent as much as possible.

A lot of the internal headers already have include guards, and #9333 did a lot to harmonize naming conventions for the internal include guards. Just looking at `crypto ssl providers`, I count 46 include guards.

```
msp@debian:~/src/openssl$ git grep -n 'ifndef.*_H$' crypto ssl providers | wc -l
46
```

Pragmas are used only twice.

```
msp@debian:~/src/openssl$ git grep -n 'pragma once' crypto ssl providers | wc -l
2
```